File: oj.uz/IOI18_doll/pas/doll.pas
unit doll;

interface

type
  array_of_longint = array of longint;

procedure create_circuit(M: longint; A: array_of_longint);

implementation

uses doll_lib;

procedure create_circuit(M: longint; A: array_of_longint);
var
  N, i, k: longint;
  C, X, Y: array of longint;
begin
  N := length(A);
  setlength(C, M + 1);
  C[0] := -1;
  for i := 1 to M do begin
    C[i] := 1;
  end;
  setlength(X, N);
  setlength(Y, N);
  for k := 0 to N - 1 do begin
    X[k] := A[k];
    Y[k] := A[k];
  end;
  answer(C, X, Y);
end;

end.

File: oj.uz/IOI18_doll/java/doll.java
public class doll {
  void create_circuit(int M, int[] A) {
    int N = A.length;
    int[] C = new int[M + 1];
    C[0] = -1;
    for (int i = 1; i <= M; ++i) {
      C[i] = 1;
    }
    int[] X = new int[N], Y = new int[N];
    for (int k = 0; k < N; ++k) {
      X[k] = Y[k] = A[k];
    }
    doll_lib.answer(C, X, Y);
  }
}
